File: isletimsistemlericalismasiJava-calismalarim/app/src/main/java/com/example/isletimsistemicalismasi/uygulamalar/notdefteri/model/NoteDatabase.java
package com.example.isletimsistemicalismasi.uygulamalar.notdefteri.model;

import android.annotation.SuppressLint;
import android.database.sqlite.SQLiteOpenHelper;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import androidx.annotation.Nullable;
import androidx.navigation.Navigation;

import java.util.ArrayList;

public class NoteDatabase extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "NotlarDB";
    private static final int DATABASE_VERSION = 1;

    public NoteDatabase(@Nullable Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {

        String CREATE_NOT = "CREATE TABLE IF NOT EXISTS Notlar (" +
                "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
                "baslik TEXT, " +
                "notMetin TEXT, " +
                "listName TEXT, " +
                "imageUrl TEXT, " +
                "color TEXT, " +
                "tarih TEXT)";
        db.execSQL(CREATE_NOT);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

        db.execSQL("DROP TABLE IF EXISTS Notlar");
        onCreate(db);
    }

    public void yeniNot(Not not) {
        SQLiteDatabase db = this.getWritableDatabase();
        String INSERT_NOT = "INSERT INTO Notlar (baslik, notMetin, listName, imageUrl, color, tarih) " +
                "VALUES ('" + not.getBaslik() + "', '" + not.getNotMetin() + "', '" +
                not.getListName() + "', '" + not.getImageUrl() + "', '" + not.getRenk() + "', '" + not.getTarih() + "')";
        db.execSQL(INSERT_NOT);
        db.close();
    }

    @SuppressLint("Range")
    public ArrayList<Not> getNotlarim() {
        SQLiteDatabase db = this.getWritableDatabase();
        String SELECT_NOT = "SELECT * FROM Notlar ORDER BY id DESC";
        Cursor cursor = db.rawQuery(SELECT_NOT, null);
        ArrayList<Not> notlar = new ArrayList<>();

           while (cursor.moveToNext()){
               Not not = new Not(
                       cursor.getInt(cursor.getColumnIndex("id")),
                       cursor.getString(cursor.getColumnIndex("baslik")),
                       cursor.getString(cursor.getColumnIndex("notMetin")),
                       cursor.getString(cursor.getColumnIndex("listName")),
                       cursor.getString(cursor.getColumnIndex("imageUrl")),
                       cursor.getString(cursor.getColumnIndex("color")),
                       cursor.getString(cursor.getColumnIndex("tarih")));
               notlar.add(not);
           }
      cursor.close();
        return notlar;
    }

    public void NotGuncelle(int id,Not not) {
        SQLiteDatabase db = this.getWritableDatabase();
        String UPDATE_NOT = "UPDATE Notlar SET baslik='" + not.getBaslik() + "', " +
                "notMetin='" + not.getNotMetin() + "', listName='" + not.getListName() + "', " +
                "imageUrl='" + not.getImageUrl() + "', color='" + not.getRenk() + "', " +
                "tarih='" + not.getTarih() + "' WHERE id=" + id;
        db.execSQL(UPDATE_NOT);
        db.close();
    }

    public void NotSil(Not not) {
        SQLiteDatabase db = this.getWritableDatabase();
        db.delete("Notlar","id=?",new String[]{String.valueOf(not.getId())});
        db.close();
    }
    public ArrayList<Not> NotAra(String aramaKelimesi){
        ArrayList<Not> notlar = new ArrayList<>();
        SQLiteDatabase db = this.getWritableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM Notlar WHERE baslik LIKE '%" + aramaKelimesi + "%'",null);
        while (cursor.moveToNext()){
            @SuppressLint("Range") Not not = new Not(
                    cursor.getInt(cursor.getColumnIndex("id")),
                    cursor.getString(cursor.getColumnIndex("baslik")),
                    cursor.getString(cursor.getColumnIndex("notMetin")),
                    cursor.getString(cursor.getColumnIndex("listName")),
                    cursor.getString(cursor.getColumnIndex("imageUrl")),
                    cursor.getString(cursor.getColumnIndex("color")),
                    cursor.getString(cursor.getColumnIndex("tarih")));
            notlar.add(not);
        }
        return notlar;
    }
}

File: isletimsistemlericalismasikotlin-calismalarim/app/src/main/java/com/example/isletimsistemlericalismasikotlin/uygulamalar/notdefteri/model/NoteDatabase.kt
package com.example.isletimsistemlericalismasikotlin.uygulamalar.notdefteri.model

import android.annotation.SuppressLint
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class NoteDatabase(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {
        private const val DATABASE_NAME = "NotlarDB"
        private const val DATABASE_VERSION = 1
    }

    override fun onCreate(db: SQLiteDatabase) {
        val createNotTable = """
            CREATE TABLE IF NOT EXISTS Notlar (
                id INTEGER PRIMARY KEY AUTOINCREMENT, 
                baslik TEXT, 
                notMetin TEXT, 
                listName TEXT, 
                imageUrl TEXT, 
                color TEXT, 
                tarih TEXT
            )
        """.trimIndent()
        db.execSQL(createNotTable)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS Notlar")
        onCreate(db)
    }

    fun yeniNot(not: Not) {
        writableDatabase.use { db ->
            val insertQuery = """
                INSERT INTO Notlar (baslik, notMetin, listName, imageUrl, color, tarih) 
                VALUES ('${not.baslik}', '${not.notMetin}', '${not.listName}', 
                '${not.imageUrl}', '${not.renk}', '${not.tarih}')
            """.trimIndent()
            db.execSQL(insertQuery)
        }
    }

    @SuppressLint("Range")
    fun getNotlarim(): ArrayList<Not> {
        val notlar = ArrayList<Not>()
        readableDatabase.use { db ->
            val cursor = db.rawQuery("SELECT * FROM Notlar ORDER BY id DESC", null)
            cursor.use {
                while (it.moveToNext()) {
                    val not = Not(
                        it.getInt(it.getColumnIndex("id")),
                        it.getString(it.getColumnIndex("baslik")),
                        it.getString(it.getColumnIndex("notMetin")),
                        it.getString(it.getColumnIndex("listName")),
                        it.getString(it.getColumnIndex("imageUrl")),
                        it.getString(it.getColumnIndex("color")),
                        it.getString(it.getColumnIndex("tarih"))
                    )
                    notlar.add(not)
                }
            }
        }
        return notlar
    }

    fun notGuncelle(id: Int, not: Not) {
        writableDatabase.use { db ->
            val updateQuery = """
                UPDATE Notlar SET baslik='${not.baslik}', notMetin='${not.notMetin}', 
                listName='${not.listName}', imageUrl='${not.imageUrl}', 
                color='${not.renk}', tarih='${not.tarih}' WHERE id=$id
            """.trimIndent()
            db.execSQL(updateQuery)
        }
    }

    fun notSil(not: Not) {
        writableDatabase.use { db ->
            db.delete("Notlar", "id=?", arrayOf(not.id.toString()))
        }
    }

    @SuppressLint("Range")
    fun notAra(aramaKelimesi: String): ArrayList<Not> {
        val notlar = ArrayList<Not>()
        readableDatabase.use { db ->
            val cursor = db.rawQuery("SELECT * FROM Notlar WHERE baslik LIKE '%$aramaKelimesi%'", null)
            cursor.use {
                while (it.moveToNext()) {
                    val not = Not(
                        it.getInt(it.getColumnIndex("id")),
                        it.getString(it.getColumnIndex("baslik")),
                        it.getString(it.getColumnIndex("notMetin")),
                        it.getString(it.getColumnIndex("listName")),
                        it.getString(it.getColumnIndex("imageUrl")),
                        it.getString(it.getColumnIndex("color")),
                        it.getString(it.getColumnIndex("tarih"))
                    )
                    notlar.add(not)
                }
            }
        }
        return notlar
    }
}
